src/query/functions/aggregation/take.go

```diff
@@ -136,8 +141,18 @@ func (n *takeNode) ProcessBlock(queryCtx *models.QueryContext, ID parser.NodeID,
 		seriesMetas,
 	)
 
-	// retain original metadatas
-	builder, err := n.controller.BlockBuilder(queryCtx, meta, stepIter.SeriesMeta())
+	seriesCount := maxSeriesCount(buckets)
+	if instantaneous {
+		heap := utils.NewFloatHeap(takeTop, utils.Min(n.op.k, seriesCount))
```

Comment:
```suggestion
		heapSize := seriesCount
		if n.op.k < seriesCount {
			heapSize = n.op.k
		}
		
		heap := utils.NewFloatHeap(takeTop, heapSize)
```

Reply:
Sorry, but I don't understand your point here. Why do you think this is better? IMO, you've just added more code and reduced readability. Now I need to read 5 more lines of code in order to understand the logic how `heapSize` is calculated. Also, by doing this you essentially made this logic more difficult to test...

Reply:
Otherwise we end up with a ton of little utility functions that never really get reused since nobody else is aware of them

Reply:
Removed `Min` function and used inline check.